core/PageTableWalker.hpp

```diff
@@ -0,0 +1,129 @@
+//
+// Created by skhan on 11/20/24.
+//
+namespace atlas
+{
+
+    struct Va32Bits
```

Comment:
By creating the struct and unions, I am abstracting the code which extracts the vpn(s), offset from the Virtual Address.
In the approach you mentioned, as far as I am understanding, it will need creation of methods which will extract vpn(s), offset but those will be called by passing the va everytime to those. Instead, my appraoch just needs a line of code for converting uint32_t virtualAddress to corresponding VA struct, and then all the code to extract VPN(s) and offset is abstracted.